packages/shared/src/features/onboarding/hooks/useStepTransition.ts

```diff
@@ -0,0 +1,43 @@
+import type { UseMutateAsyncFunction } from '@tanstack/react-query';
+import { useMutation } from '@tanstack/react-query';
+import { HttpError } from '../../../lib/errors';
+
+export type StepInputs = Record<string, unknown>;
+
+export interface StepTransitionPayload {
+  fromStep: string;
+  toStep: string | null;
+  transitionEvent: string;
```

Comment:
Maybe we can use `FunnelStepTransitionType`?

Reply:
done